screen2.0/src/app/search/search-helpers.ts

```diff
@@ -266,37 +452,178 @@ export function constructURL(
+      dnase_e: searchParams.dnase_e ? Number(searchParams.dnase_e) : 10,
+      atac_s: searchParams.atac_s ? Number(searchParams.atac_s) : -10,
+      atac_e: searchParams.atac_e ? Number(searchParams.atac_e) : 10,
+      h3k4me3_s: searchParams.h3k4me3_s ? Number(searchParams.h3k4me3_s) : -10,
+      h3k4me3_e: searchParams.h3k4me3_e ? Number(searchParams.h3k4me3_e) : 10,
+      h3k27ac_s: searchParams.h3k27ac_s ? Number(searchParams.h3k27ac_s) : -10,
+      h3k27ac_e: searchParams.h3k27ac_e ? Number(searchParams.h3k27ac_e) : 10,
+      ctcf_s: searchParams.ctcf_s ? Number(searchParams.ctcf_s) : -10,
+      ctcf_e: searchParams.ctcf_e ? Number(searchParams.ctcf_e) : 10,
+      prim_s: searchParams.prim_s ? Number(searchParams.prim_s) : -2,
+      prim_e: searchParams.prim_e ? Number(searchParams.prim_e) : 2,
+      mamm_s: searchParams.mamm_s ? Number(searchParams.mamm_s) : -4,
+      mamm_e: searchParams.mamm_e ? Number(searchParams.mamm_e) : 8,
+      vert_s: searchParams.vert_s ? Number(searchParams.vert_s) : -3,
+      vert_e: searchParams.vert_e ? Number(searchParams.vert_e) : 8,
+      CA: searchParams.CA ? checkTrueFalse(searchParams.CA) : true,
+      CA_CTCF: searchParams.CA_CTCF ? checkTrueFalse(searchParams.CA_CTCF) : true,
+      CA_H3K4me3: searchParams.CA_H3K4me3 ? checkTrueFalse(searchParams.CA_H3K4me3) : true,
+      CA_TF: searchParams.CA_TF ? checkTrueFalse(searchParams.CA_TF) : true,
+      dELS: searchParams.dELS ? checkTrueFalse(searchParams.dELS) : true,
+      pELS: searchParams.pELS ? checkTrueFalse(searchParams.pELS) : true,
+      PLS: searchParams.PLS ? checkTrueFalse(searchParams.PLS) : true,
+      TF: searchParams.TF ? checkTrueFalse(searchParams.TF) : true,
+      genesToFind: searchParams.genesToFind ? searchParams.genesToFind.split(",") : [],
+      distancePC: searchParams.distancePC ? checkTrueFalse(searchParams.distancePC) : true,
+      distanceAll: searchParams.distanceAll ? checkTrueFalse(searchParams.distanceAll) : true,
+      CTCF_ChIA_PET: searchParams.CTCF_ChIA_PET ? checkTrueFalse(searchParams.CTCF_ChIA_PET) : true,
+      RNAPII_ChIA_PET: searchParams.RNAPII_ChIA_PET ? checkTrueFalse(searchParams.RNAPII_ChIA_PET) : true
+    }
+  )
+}
+
+export function constructBiosampleTableFiltersFromURL(searchParams: { [key: string]: string | undefined }): BiosampleTableFilters {
+  return (
+    {
+      CellLine: searchParams.CellLine ? checkTrueFalse(searchParams.CellLine) : true,
+      PrimaryCell: searchParams.PrimaryCell ? checkTrueFalse(searchParams.PrimaryCell) : true,
+      Tissue: searchParams.Tissue ? checkTrueFalse(searchParams.Tissue) : true,
+      Organoid: searchParams.Organoid ? checkTrueFalse(searchParams.Organoid) : true,
+      InVitro: searchParams.InVitro ? checkTrueFalse(searchParams.InVitro) : true,
+    }
+  )
+}
+
+export function createQueryString(searchParams: { [key: string]: string | undefined }, name1: string, value1: string, name2?: string, value2?: string){
```

Comment:
input parameter can be array of name , value pair rather than using name1 and name2

Reply:
fixed, function wasn't being used so removed